rust/src/ffi.rs

```diff
@@ -1271,3 +1271,84 @@ pub unsafe extern "C" fn zcashlc_free_address_check_result(ptr: *mut AddressChec
         drop(res)
     }
 }
+
+/// A received transaction output from any pool (transparent, sapling, orchard).
+#[repr(C)]
+pub struct ReceivedOutput {
+    /// Pool type (0 = transparent, 1 = sapling, 2 = orchard)
+    pool_type: u32,
+    /// Output index within the transaction
+    output_index: usize,
+    /// Value in zatoshis
+    value: u64,
+    /// Number of confirmations until spendable
+    confirmations_until_spendable: u32,
+}
+
+impl ReceivedOutput {
+    pub(crate) fn from_received_transaction_output(
+        output: &zcash_client_backend::data_api::ReceivedTransactionOutput,
+    ) -> Self {
+        use zcash_protocol::PoolType;
+
+        let pool_type = match output.pool_type() {
+            PoolType::Transparent => 0,
+            PoolType::Shielded(zcash_protocol::ShieldedProtocol::Sapling) => 1,
+            PoolType::Shielded(zcash_protocol::ShieldedProtocol::Orchard) => 2,
+        };
```

Comment:
Footgun: this should match how we pass protocol types into `zcashlc_get_memo`, both for FFI consistency, and because otherwise if the existing Swift type is used it would be incorrect. And that was based off the UA typecodes:
https://github.com/Electric-Coin-Company/zcash-light-client-ffi/blob/d373b2e501d33396900231de709c43f01fd3424a/rust/src/lib.rs#L1195-L1201

```suggestion
        let pool_type = match output.pool_type() {
            PoolType::Transparent => 0,
            PoolType::Shielded(zcash_protocol::ShieldedProtocol::Sapling) => 2,
            PoolType::Shielded(zcash_protocol::ShieldedProtocol::Orchard) => 3,
        };
```